pkgs/development/libraries/gaia/default.nix

```diff
@@ -0,0 +1,81 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, libyaml
+, swig
+, pkgconfig
+, wafHook
+, makeWrapper
+, qt4
+, python
+, pythonSupport ? true
+# Default to false since it brakes the build
+, stlfacadeSupport ? false
+, assertsSupport ? true
+, cyclopsSupport ? true
+}:
+
+assert pythonSupport -> python != null;
+
+stdenv.mkDerivation rec {
+  pname = "gaia";
+  version = "2.4.5";
+
+  src = fetchFromGitHub {
+    owner = "MTG";
+    repo = "gaia";
+    rev = "v${version}";
+    sha256 = "12jxb354s2dblr2ghnl3w05m23jgzvrrgywfj8jaa32j3gw48fv2";
+  };
+
+  # Fix installation error when waf tries to put files in /etc/
+  prePatch = ''
+  '' + lib.optionalString cyclopsSupport ''
+    substituteInPlace src/wscript \
+      --replace "/etc/cyclops" "$out/etc/cyclops" \
+      --replace "/etc/init.d" "$out/etc/init.d"
+  '';
+
+  # This is not exactly specified in upstream's README but it's needed by the
+  # resultings $out/bin/gaiafusion script
+  pythonEnv = (if pythonSupport then
+    python.withPackages(ps: with ps; [
+      pyyaml
+    ])
+  else null);
+
+  nativeBuildInputs = [
+    wafHook
+    pkgconfig
+    swig
+    makeWrapper
+  ];
+  buildInputs = [
+    libyaml
+    qt4
+  ]
+    ++ lib.optionals (pythonSupport) [
+      pythonEnv
+    ]
+  ;
+  wafConfigureFlags = [
+  ]
+    ++ lib.optionals (pythonSupport) [ "--with-python-bindings" ]
```

Comment:
As mentioned in https://github.com/NixOS/nixpkgs/pull/70389/files#r331696607, use of `lib.optional` here will already imply a singleton list.

Reply:
I've had comments in the past saying that `lib.optionals` should always be preferred over `lib.optional`. See https://github.com/NixOS/nixpkgs/pull/65480#discussion_r315800912

Reply:
Again, similar to the superfluous empty lists/strings it's just personal opinion, but I do not find `lib.optional` a bad thing, because the less scanning one needs to do for `[` and `]` the better IMO. But again, it's a matter of personal preference, and I also tend to use `lib.singleton` a lot, where other people would probably shriek in horror ;-)

Reply:
I see, my main motive is that future diffs will be clean and I'd like at least to be consistent with my self.. I appreciate your opinion though. Allow me to mark this conversation as resolved based upon our agreement upon disagreement :relaxed: .